cmd/sls/sls_hardware.go

```diff
@@ -479,7 +479,7 @@ func doHardwareSearch(w http.ResponseWriter, r *http.Request) {
 				pdet := base.NewProblemDetails("about: blank",
 					"Internal Server Error",
 					"Failed to search hardware in DB. ExtraProperties search does not include field.",
-					r.URL.Path, http.StatusInternalServerError)
+					r.URL.Path, http.StatusBadRequest)
```

Comment:
help me understand the process that causes us to get here.

Reply:
Here is a successful "extra properties" search:
`curl -s -v -X GET 'localhost:8376/v1/search/hardware?extra_properties.Brand=Mellanox'`

The failure case is hit with this call (which now returns 400 instead of 500)
`curl -s -v -X GET 'localhost:8376/v1/search/hardware?extra_properties.=Mellanox'`
The failure case seems unlikely to be hit. Possibly it should also return the error for the following case (without the period)
`curl -s -v -X GET 'localhost:8376/v1/search/hardware?extra_properties=Mellanox'`

Currently this case returns:
`{"type":"about: blank","title":"Bad Request","detail":"no conditions/properties with which to search","instance":"/v1/search/hardware","status":400}`

Reply:
Much better; 👍🏻